collections/src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -288,7 +288,7 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] {
     *
     *  @return    `true` if the $coll contains at least one element, `false` otherwise.
     */
-  def nonEmpty: Boolean = iterator().hasNext
+  def nonEmpty: Boolean = !isEmpty
```

Comment:
> Implementing operations in terms of others can be confusing because then subclasses might override the wrong one

Yes, that's precisely the motivation. It was defined like this in 2.12 but without this change you would have to override *both* `nonEmpty` and `isEmpty` (which no existing collections do) which is never what you want.

Reply:
We can mark it `@deprecatedOverriding` for now and maybe make it `@inline final` later (but I doubt it makes much of a difference because the JIT can inline it anyway)

Reply:
I'm in favor of marking it final now, it's simple enough to fix custom subclasses, and this is hardly going to be the only thing that is source-incompatible.